app-backend/export/src/main/scala/com/azavea/rf/export/Export.scala

```diff
@@ -0,0 +1,146 @@
+package com.azavea.rf.export
+
+import com.azavea.rf.export.model._
+import com.azavea.rf.export.util._
+
+import geotrellis.proj4.LatLng
+import geotrellis.raster._
+import geotrellis.raster.io._
+import geotrellis.raster.histogram.Histogram
+import geotrellis.raster.io.geotiff.{GeoTiff, MultibandGeoTiff}
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.tiling._
+import geotrellis.spark.io.hadoop._
+import geotrellis.spark.io.file._
+import geotrellis.spark.io.s3._
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.fs.Path
+import org.apache.spark._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import cats.implicits._
+
+import java.util.UUID
+
+object Export extends SparkJob with LazyLogging {
+  /** Get a LayerReader and an attribute store for the catalog located at the provided URI
+    *
+    * @param exportLayerDef export job's layer definition
+    */
+  def getRfLayerManagement(exportLayerDef: ExportLayerDefinition)(implicit @transient sc: SparkContext): (FilteringLayerReader[LayerId], AttributeStore) =
+    exportLayerDef.ingestLocation.getScheme match {
+      case "s3" | "s3a" | "s3n" =>
+        val (bucket, prefix) = S3.parse(exportLayerDef.ingestLocation)
+        val reader = S3LayerReader(bucket, prefix)
+        (reader, reader.attributeStore)
+      case "file" =>
+        val reader = FileLayerReader(exportLayerDef.ingestLocation.getPath)
+        (reader, reader.attributeStore)
+    }
+
+  def exportProject(params: CommandLine.Params)(exportDefinition: ExportDefinition)(implicit @transient sc: SparkContext) = {
+    val wrappedConfiguration = HadoopConfiguration(S3.setCredentials(sc.hadoopConfiguration))
+    val (input, output) = exportDefinition.input -> exportDefinition.output
+    val rdds = input.layers.map { ld =>
+      val (reader, attributeStore) = getRfLayerManagement(ld)
+      val layerId = LayerId(ld.layerId.toString, input.resolution)
+      val md = attributeStore.readMetadata[TileLayerMetadata[SpatialKey]](layerId)
+      val hist = ld.colorCorrections.map { _ => attributeStore.read[Array[Histogram[Double]]](LayerId(layerId.name, 0), "histogram") }
+      val crs = output.crs.getOrElse(md.crs)
+
+      val query = {
+        val q = reader.query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](layerId)
+        input.mask.fold(q)(mp => q.where(Intersects(mp.reproject(LatLng, md.crs))))
+      } result
```

Comment:
Querying for scene which does not have required resolution would fail here. The tiler would find the closest available zoom and up-sample it. This behavior is also desirable in the export as its supposed to mirror the tiler but I think it should be implemented in subsequent PRs, just calling it out here.